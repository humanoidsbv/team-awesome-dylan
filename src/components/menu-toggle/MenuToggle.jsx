import React from 'react';
import PropTypes from 'prop-types';

import styles from './MenuToggle.module.css';

const MenuToggle = ({ toggleMenu, isMenuVisible }) => (
  <button onClick={toggleMenu} className={styles.toggle}>
    <span className={`${styles.icon} ${isMenuVisible ? styles.hideIcon : ''}`}>
      =
    </span>
    <span className={`${styles.icon} ${isMenuVisible ? '' : styles.hideIcon}`}>
      x
    </span>
  </button>
);

MenuToggle.propTypes = {
  toggleMenu: PropTypes.func.isRequired,
  isMenuVisible: PropTypes.bool.isRequired,
};

export default MenuToggle;
