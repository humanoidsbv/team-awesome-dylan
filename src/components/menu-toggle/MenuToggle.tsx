import React from 'react';

import IconClose from '../../assets/icons/icon-close.svg';
import IconMenu from '../../assets/icons/icon-menu.svg';

import styles from './MenuToggle.module.css';

interface MenuToggleInterface {
  toggleMenu: () => void;
  isMenuVisible: boolean;
}

const MenuToggle = ({ toggleMenu, isMenuVisible }: MenuToggleInterface): React.ReactElement => (
  <button type="button" onClick={toggleMenu} className={styles.toggle}>
    <IconMenu
      className={`${styles.icon} ${isMenuVisible ? styles.hideIcon : ''}`}
    />
    <IconClose
      className={`${styles.icon} ${isMenuVisible ? '' : styles.hideIcon}`}
    />
  </button>
);

export default MenuToggle;
