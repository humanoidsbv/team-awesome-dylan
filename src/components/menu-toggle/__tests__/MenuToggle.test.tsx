import * as React from 'react';
import { shallow } from 'enzyme';

import MenuToggle from '../MenuToggle';

it('matches snapshot', () => {
  const menuToggle = shallow(
    <MenuToggle
    toggleMenu={() => null}
    isMenuVisible= {true}
    />
  );
  expect(menuToggle).toMatchSnapshot();
});
