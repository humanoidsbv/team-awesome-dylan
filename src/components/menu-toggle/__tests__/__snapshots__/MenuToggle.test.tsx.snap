// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="toggle"
  onClick={[Function]}
  type="button"
>
  <test-file-stub
    className="icon hideIcon"
  />
  <test-file-stub
    className="icon "
  />
</button>
`;
